### PR TITLE
Migrating repo to use setuptools

### PR DESCRIPTION
**Pull request by NishanthJKumar**
This PR introduces a sweeping set of organizational changes to make the repository use `setuptools` for installation and dependency tracking. This has the side benefit of making it easy to have multiple forks of the same repository on the same machine/under the same organization.

### CODE REVIEW

**Comment by tomsilver** [comment]
we should change the name of this PR lol

File: README.md [hunk 1]
```diff
@@ -39,10 +38,10 @@ See [these instructions](supercloud.md).
...
 ## Instructions For Contributing
-* Run `pip install -r requirements-dev.txt` to install all dependencies for development.
+* Run `pip install -e .[develop]` to install all dependencies for development.
 * You can't push directly to master. Make a new branch in this repository (don't use a fork, since that will not properly trigger the checks when you make a PR). When your code is ready for review, make a PR and request reviews from the appropriate people.
 * To merge a PR, you need at least one approval, and you have to pass the 4 checks defined in `.github/workflows/predicators.yml`, which you can run locally in one line via `./scripts/run_checks.sh`, or individually as follows:
-    * `pytest -s tests/ --cov-config=.coveragerc --cov=src/ --cov=tests/ --cov-fail-under=100 --cov-report=term-missing:skip-covered --durations=0`
+    * `pytest -s tests/ --cov-config=.coveragerc --cov=predicators/ --cov=tests/ --cov-fail-under=100 --cov-report=term-missing:skip-covered --durations=0`
     * `mypy . --config-file mypy.ini`
     * `pytest . --pylint -m pylint --pylint-rcfile=.predicators_pylintrc`
     * `yapf -i -r --style .style.yapf . && docformatter -i -r . && isort .`
```
Comment (tomsilver) [inline README.md:47]: change this to `./run_autoformat.sh`

File: setup.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
           "numpy>=1.22.3", "pytest", "gym==0.21.0", "matplotlib", "imageio",
           "imageio-ffmpeg", "pandas", "torch", "scipy", "tabulate", "dill",
           "pyperplan", "pathos", "requests", "slack_bolt", "pybullet>=3.2.0",
-          "sklearn", "pyqt5", "graphlib-backport", "openai", "pyyaml",
+          "sklearn", "graphlib-backport", "openai", "pyyaml",
```
Comment (tomsilver) [inline setup.py]: pyqt5 is only needed for collecting human demonstrations. this isn't actively being used. it's a flaming disaster to install on macs with m1 chips. so i'm just removing it for now

File: scripts/plotting/create_num_demos_plots.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import matplotlib.pyplot as plt
```
Comment (tomsilver) [inline scripts/plotting/create_num_demos_plots.py:6]: I'm attempting to run some of these scripts with old results. Unfortunately it looks like we're going to have pickling issues if we try to do that:
```
Traceback (most recent call last):
  File "/Users/tom/phd/predicators/scripts/plotting/create_num_demos_plots.py", line 150, in <module>
    _main()
  File "/Users/tom/phd/predicators/scripts/plotting/create_num_demos_plots.py", line 112, in _main
    grouped_means, grouped_stds, _ = create_dataframes(COLUMN_NAMES_AND_KEYS,
  File "/Users/tom/phd/predicators/scripts/analyze_results_directory.py", line 159, in create_dataframes
    df = create_raw_dataframe(column_names_and_keys, derived_keys)
  File "/Users/tom/phd/predicators/scripts/analyze_results_directory.py", line 109, in create_raw_dataframe
    outdata = pkl.load(f)
  File "/Users/tom/.pyenv/versions/3.9.4/lib/python3.9/site-packages/dill/_dill.py", line 373, in load
    return Unpickler(file, ignore=ignore, **kwds).load()
  File "/Users/tom/.pyenv/versions/3.9.4/lib/python3.9/site-packages/dill/_dill.py", line 646, in load
    obj = StockUnpickler.load(self)
  File "/Users/tom/.pyenv/versions/3.9.4/lib/python3.9/site-packages/dill/_dill.py", line 636, in find_class
    return StockUnpickler.find_class(self, module, name)
ModuleNotFoundError: No module named 'predicators.src'
```

I think we can live with this, people will just have to check out an old version if they want to use old results... and everyone is more or less in between projects right now, so it's good that we're making this change now.

but, we should include a warning about this in the announcement that we'll make about this change.

**Review by tomsilver — COMMENTED** [review]
okay, I think this is pretty much ready to go, modulo the announcement! @NishanthJKumar can you draft the announcement and then we'll test it out on someone (maybe @wmcclinton ) before sharing it with everyone else? the hope is that it contains sufficient information for everyone to switch over their local setups

File: scripts/plotting/create_num_demos_plots.py (hunk 3 above)
Comment (NishanthJKumar) [reply scripts/plotting/create_num_demos_plots.py:6]: ahhh crap - i see. Well, making releases for all our major papers was a really good idea then!

**Review by tomsilver — APPROVED** [review]
@wmcclinton tested too and it looks like we're ready!

**Pull request merged**